### PR TITLE
Initial CRI-O support

### PR DESCRIPTION
**Pull request by JAORMX**
This takes modifies udica to also take the "inspect" format that crictl
gives out, and not only the docker/podman one.

Note that in this implementation, only the json file works; support for
parsing the input from the crictl command will come in a separate PR.

### CODE REVIEW

**Comment by JAORMX** [comment]
Didn't reaaaally test extensively if it still works with podman/docker. So... if someone could help verify that it would be great :smile:

**Comment by JAORMX** [comment]
@wrabcak could you take a look?

**Comment by JAORMX** [comment]
the meta task seems broken with an unrelated error.

File: udica/policy.py [hunk 1]
```diff
@@ -138,6 +138,74 @@ def create_policy(opts, capabilities, mounts, ports, append_rules):
...
+
+        if item['hostPath'] == CONFIG_CONTAINER:
+            if item['readonly']:
+                policy.write('    (blockinherit config_container)\n')
+            else:
+                policy.write('    (blockinherit config_rw_container)\n')
+            add_template("config_container")
+            continue
+
+        # TODO(jaosorior): Add prefix-dir to path. This way we could call this
```
Comment (wrabcak) [inline udica/policy.py]: Would be possible to implement this TODO? I'm afraid that we forget about it, if there is just TODO liner in code. :)

File: udica/policy.py [hunk 2]
```diff
@@ -138,6 +138,74 @@ def create_policy(opts, capabilities, mounts, ports, append_rules):
...
+        # from a container in kubernetes
+        contexts = list_contexts(item['hostPath'])
+        for context in contexts:
+            if item['readonly']:
```
Comment (wrabcak) [inline udica/policy.py]: I think, here is a bug. In If statement you're checking if the mount point is "readonly", and if it's true, you are using "drw/frw/srw" permission sets. It should be switched, there should be "dro/fro/sro".

**Comment by wrabcak** [comment]
@JAORMX , 
We also need need tests for this new feature. Could you please share inspection file from CRI-O engine, where you're mounting /home with read-only perms and /var/spool/ with read/write perms? Something similar to  https://github.com/containers/udica/blob/master/tests/test_basic.podman.json and I'll add some tests to upstream testsuite. 
Thanks,
Lukas.

**Comment by AkihiroSuda** [comment]
Is this specific to crio?
Does it work with containerd as well?

**Comment by JAORMX** [comment]
@AkihiroSuda I only tested it on CRI-O. But in theory it should work on containerd as well, since they all follow the CRI interface. Would be a matter of testing it out.

**Comment by wrabcak** [comment]
Merging, Thanks!

**Pull request merged**